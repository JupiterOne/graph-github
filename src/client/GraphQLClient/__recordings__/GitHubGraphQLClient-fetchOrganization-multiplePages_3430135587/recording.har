{
  "log": {
    "_recordingName": "GitHubGraphQLClient.fetchOrganization.multiplePages",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "7f2926ac1ad814d2dfcaf9525a63bf36",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.machine-man-preview+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/17.6.0 octokit-core.js/2.5.4 Node.js/10.19.0 (Linux 4.4; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "Bearer [REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 815,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "queryString": [],
          "url": "https://api.github.com/app/installations/17214088/access_tokens"
        },
        "response": {
          "bodySize": 575,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 575,
            "text": "{\"token\":\"[REDACTED]\",\"expires_at\":\"2021-06-04T19:59:18Z\",\"permissions\":{\"members\":\"read\",\"organization_packages\":\"write\",\"actions\":\"write\",\"administration\":\"write\",\"checks\":\"write\",\"contents\":\"write\",\"deployments\":\"write\",\"discussions\":\"write\",\"environments\":\"write\",\"issues\":\"write\",\"metadata\":\"read\",\"packages\":\"write\",\"pages\":\"write\",\"pull_requests\":\"write\",\"repository_hooks\":\"write\",\"repository_projects\":\"admin\",\"secrets\":\"write\",\"secret_scanning_alerts\":\"write\",\"security_events\":\"write\",\"statuses\":\"write\"},\"repository_selection\":\"all\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Fri, 04 Jun 2021 18:59:19 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "575"
            },
            {
              "name": "cache-control",
              "value": "public, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "etag",
              "value": "\"2b5dc60a9a5ad5850a829605a4749d60a36562aa23c8fd8e48cf38d1af36d95d\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.v3; param=machine-man-preview; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "x-github-request-id",
              "value": "F89A:1BDF:1619CBA:2B27F91:60BA7806"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1038,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 201,
          "statusText": "Created"
        },
        "startedDateTime": "2021-06-04T18:59:19.298Z",
        "time": 306,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 306
        }
      },
      {
        "_id": "22e4c882794d943e8a6f0961702d5000",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 831,
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "user-agent",
              "value": "jupiterone-graph-github"
            },
            {
              "name": "authorization",
              "value": "Bearer [REDACTED]"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 228,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "{\"query\":\"query ($login: String!, $members: String) {\\n    organization(login: $login) {\\n        id\\n        ... organizationFields\\n        membersWithRole(first: 2, after: $members) {\\n        edges {\\n          node {\\n            id\\n            ... userFields\\n            \\n          }\\n\\n          ... userEdgeFields\\n        }\\n\\n        pageInfo {\\n  endCursor\\n  hasNextPage\\n}\\n      }\\n      }\\n... rateLimit\\n  }\\n\\nfragment organizationFields on Organization {\\n    login\\n    name\\n  }\\n\\nfragment userFields on User {\\n    login\\n    name\\n    isSiteAdmin\\n  }\\n\\nfragment userEdgeFields on OrganizationMemberEdge {\\n    hasTwoFactorEnabled\\n    role\\n  }\\n\\nfragment rateLimit on Query {\\n    rateLimit {\\n      limit\\n      cost\\n      remaining\\n      resetAt\\n    }\\n  }\",\"variables\":{\"login\":\"Kei-Institute\"}}"
          },
          "queryString": [],
          "url": "https://api.github.com/graphql"
        },
        "response": {
          "bodySize": 587,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 587,
            "text": "{\"data\":{\"organization\":{\"id\":\"MDEyOk9yZ2FuaXphdGlvbjg0OTIzNTAz\",\"login\":\"Kei-Institute\",\"name\":\"Kei-Institute\",\"membersWithRole\":{\"edges\":[{\"node\":{\"id\":\"MDQ6VXNlcjUxMzUyMw==\",\"login\":\"erichs\",\"name\":\"Erich Smith\",\"isSiteAdmin\":false},\"hasTwoFactorEnabled\":null,\"role\":\"MEMBER\"},{\"node\":{\"id\":\"MDQ6VXNlcjI1NDg5NDgy\",\"login\":\"mknoedel\",\"name\":\"Michael Knoedel\",\"isSiteAdmin\":false},\"hasTwoFactorEnabled\":null,\"role\":\"MEMBER\"}],\"pageInfo\":{\"endCursor\":\"Y3Vyc29yOnYyOpHOAYTwSg==\",\"hasNextPage\":true}}},\"rateLimit\":{\"limit\":5000,\"cost\":1,\"remaining\":4983,\"resetAt\":\"2021-06-04T19:49:50Z\"}}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Fri, 04 Jun 2021 18:59:19 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "587"
            },
            {
              "name": "cache-control",
              "value": "no-cache"
            },
            {
              "name": "x-github-media-type",
              "value": "github.v4"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "4983"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1622836190"
            },
            {
              "name": "x-ratelimit-used",
              "value": "17"
            },
            {
              "name": "x-ratelimit-resource",
              "value": "graphql"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "x-github-request-id",
              "value": "F89B:3AAC:C5D96C:1C60696:60BA7807"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1030,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-04T18:59:19.614Z",
        "time": 379,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 379
        }
      },
      {
        "_id": "22e4c882794d943e8a6f0961702d5000",
        "_order": 1,
        "cache": {},
        "request": {
          "bodySize": 868,
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "user-agent",
              "value": "jupiterone-graph-github"
            },
            {
              "name": "authorization",
              "value": "Bearer [REDACTED]"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 228,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "{\"query\":\"query ($login: String!, $members: String) {\\n    organization(login: $login) {\\n        id\\n        ... organizationFields\\n        membersWithRole(first: 2, after: $members) {\\n        edges {\\n          node {\\n            id\\n            ... userFields\\n            \\n          }\\n\\n          ... userEdgeFields\\n        }\\n\\n        pageInfo {\\n  endCursor\\n  hasNextPage\\n}\\n      }\\n      }\\n... rateLimit\\n  }\\n\\nfragment organizationFields on Organization {\\n    login\\n    name\\n  }\\n\\nfragment userFields on User {\\n    login\\n    name\\n    isSiteAdmin\\n  }\\n\\nfragment userEdgeFields on OrganizationMemberEdge {\\n    hasTwoFactorEnabled\\n    role\\n  }\\n\\nfragment rateLimit on Query {\\n    rateLimit {\\n      limit\\n      cost\\n      remaining\\n      resetAt\\n    }\\n  }\",\"variables\":{\"login\":\"Kei-Institute\",\"members\":\"Y3Vyc29yOnYyOpHOAYTwSg==\"}}"
          },
          "queryString": [],
          "url": "https://api.github.com/graphql"
        },
        "response": {
          "bodySize": 449,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 449,
            "text": "{\"data\":{\"organization\":{\"id\":\"MDEyOk9yZ2FuaXphdGlvbjg0OTIzNTAz\",\"login\":\"Kei-Institute\",\"name\":\"Kei-Institute\",\"membersWithRole\":{\"edges\":[{\"node\":{\"id\":\"MDQ6VXNlcjYyNDkyMDk3\",\"login\":\"kevincasey1222\",\"name\":\"Kevin Casey\",\"isSiteAdmin\":false},\"hasTwoFactorEnabled\":null,\"role\":\"ADMIN\"}],\"pageInfo\":{\"endCursor\":\"Y3Vyc29yOnYyOpHOA7mNwQ==\",\"hasNextPage\":false}}},\"rateLimit\":{\"limit\":5000,\"cost\":1,\"remaining\":4982,\"resetAt\":\"2021-06-04T19:49:50Z\"}}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Fri, 04 Jun 2021 18:59:19 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "449"
            },
            {
              "name": "cache-control",
              "value": "no-cache"
            },
            {
              "name": "x-github-media-type",
              "value": "github.v4"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "4982"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1622836190"
            },
            {
              "name": "x-ratelimit-used",
              "value": "18"
            },
            {
              "name": "x-ratelimit-resource",
              "value": "graphql"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding, Accept, X-Requested-With"
            },
            {
              "name": "x-github-request-id",
              "value": "F89C:43C0:C4512C:1BDB002:60BA7807"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 1030,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-04T18:59:20.001Z",
        "time": 366,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 366
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
