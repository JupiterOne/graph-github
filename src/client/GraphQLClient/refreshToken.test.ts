import {
  createMockStepExecutionContext,
  Recording,
  createMockIntegrationLogger,
} from '@jupiterone/integration-sdk-testing';
import { setupGithubRecording } from '../../../test/recording';
import { GitHubGraphQLClient, GithubResource } from '.';
import resourceMetadataMap from './resourceMetadataMap';
import createGitHubAppClient from '../../util/createGitHubAppClient';
import { IntegrationConfig, sanitizeConfig } from '../../config';
import { integrationConfig } from '../../../test/config';
import { PullRequest } from './types';
import {
  PULL_REQUESTS_QUERY_STRING,
  REPOS_QUERY_STRING,
  TEAM_MEMBERS_QUERY_STRING,
  USERS_QUERY_STRING,
} from './queries';
import { parseTimePropertyValue } from '@jupiterone/integration-sdk-core';

//tests in this file similar to those in client.test.ts, except
//that they start with an expired token. You can see that refresh worked
//because there are two calls for an access token in each polly recording,
//recorded one second apart
//
//These tests were recorded later and with slightly different test data
//from the recordings made in client.test.ts, so some values passed to
//expect items are different

async function getAccess() {
  const context = createMockStepExecutionContext<IntegrationConfig>({
    instanceConfig: integrationConfig,
  });

  const config = context.instance.config;
  sanitizeConfig(config);
  //the installid in the recordings
  config.installationId = 17214088; //this is the id the recordings are under

  const appClient = createGitHubAppClient(
    config,
    createMockIntegrationLogger(),
  );
  const { token, expiresAt } = (await appClient.auth({
    type: 'installation',
  })) as {
    token: string;
    expiresAt: string;
  };
  const tokenExpires = parseTimePropertyValue(expiresAt) || 0;
  return { token, tokenExpires, appClient };
}

const pageLimit = 2;
async function getClient() {
  const { token, tokenExpires, appClient } = await getAccess();
  return new GitHubGraphQLClient(
    token,
    tokenExpires * 0, //making this be time zero simulates an expired token and forces a refresh
    resourceMetadataMap(pageLimit, pageLimit),
    createMockIntegrationLogger(),
    appClient,
  );
}

describe('pull request resources', () => {
  let p: Recording; //p for polly

  afterEach(async () => {
    await p.stop();
  });

  test('pullRequest pagination only', async () => {
    p = setupGithubRecording({
      directory: __dirname,
      name: 'GitHubGraphQLClient.refreshTokenInFetchPRs.singlePage',
    });
    const client = await getClient();

    const query = 'is:pr repo:Kei-Institute/Test-repo is:open';
    const pullRequests: PullRequest[] = [];
    const response = await client.iteratePullRequests(
      PULL_REQUESTS_QUERY_STRING,
      query,
      [GithubResource.Commits, GithubResource.Reviews, GithubResource.Labels],
      (pr) => {
        pullRequests.push(pr);
      },
    );
    expect(response.rateLimitConsumed).toEqual(1);
    expect(pullRequests.length).toEqual(2);
    expect(pullRequests).toMatchSnapshot();
  });
});

describe('organization resources', () => {
  let p: Recording; //p for polly

  afterEach(async () => {
    await p.stop();
  });

  test('single page', async () => {
    p = setupGithubRecording({
      directory: __dirname,
      name: 'GitHubGraphQLClient.refreshTokenInFetchOrg.singlePage',
      options: {
        matchRequestsBy: {
          headers: false, //must not set order:false
        },
      },
    });
    const client = await getClient();

    const data = await client.fetchFromSingle(
      REPOS_QUERY_STRING,
      GithubResource.Organization,
      [GithubResource.Repositories],
      { login: 'Kei-Institute' },
    );

    expect(data.organization).toBeDefined();
    expect(data.membersWithRole).toBeUndefined();
    expect(data.repositories).toHaveLength(3);
    expect(data.teams).toBeUndefined();
    expect(data.members).toBeUndefined();
    expect(data.rateLimitConsumed).toBe(1);
  });

  test('multiple pages', async () => {
    p = setupGithubRecording({
      directory: __dirname,
      name: 'GitHubGraphQLClient.refreshTokenInFetchOrg.multiplePages',
      options: {
        matchRequestsBy: {
          headers: false, //must not set order:false
        },
      },
    });
    const client = await getClient();

    const data = await client.fetchFromSingle(
      USERS_QUERY_STRING,
      GithubResource.Organization,
      [GithubResource.OrganizationMembers],
      { login: 'Kei-Institute' },
    );

    expect(data.organization).toBeDefined();
    expect(data.membersWithRole).toHaveLength(3);
    expect(data.repositories).toBeUndefined();
    expect(data.teams).toBeUndefined();
    expect(data.members).toBeUndefined();
    expect(data.rateLimitConsumed).toBe(1);
  });

  test('child resource only', async () => {
    p = setupGithubRecording({
      directory: __dirname,
      name: 'GitHubGraphQLClient.refreshTokenInFetchOrg.childOnly',
      options: {
        matchRequestsBy: {
          headers: false, //must not set order:false
        },
      },
    });
    const client = await getClient();

    const data = await client.fetchFromSingle(
      TEAM_MEMBERS_QUERY_STRING,
      GithubResource.Organization,
      [GithubResource.TeamMembers],
      { login: 'Kei-Institute' },
    );

    expect(data.organization).toHaveLength(1);
    expect(data.membersWithRole).toBeUndefined();
    expect(data.repositories).toBeUndefined();
    expect(data.members).toHaveLength(6);
    expect(data.members).toEqual([
      {
        id: 'MDQ6VXNlcjUxMzUyMw==',
        login: 'erichs',
        name: 'Erich Smith',
        node: undefined,
        teams: 'MDQ6VGVhbTQ4NTgxNjk=',
        role: 'MEMBER',
      },
      {
        id: 'MDQ6VXNlcjI1NDg5NDgy',
        login: 'mknoedel',
        name: 'Michael Knoedel',
        node: undefined,
        teams: 'MDQ6VGVhbTQ4NTgxNjk=',
        role: 'MEMBER',
      },
      {
        id: 'MDQ6VXNlcjYyNDkyMDk3',
        login: 'kevincasey1222',
        name: 'Kevin Casey',
        node: undefined,
        teams: 'MDQ6VGVhbTQ4NTgxNjk=',
        role: 'MAINTAINER',
      },
      {
        id: 'MDQ6VXNlcjYyNDkyMDk3',
        login: 'kevincasey1222',
        name: 'Kevin Casey',
        node: undefined,
        teams: 'MDQ6VGVhbTQ4NTgxNzA=',
        role: 'MAINTAINER',
      },
      {
        id: 'MDQ6VXNlcjI1NDg5NDgy',
        login: 'mknoedel',
        name: 'Michael Knoedel',
        node: undefined,
        teams: 'MDQ6VGVhbTQ4NTc0OTU=',
        role: 'MEMBER',
      },
      {
        id: 'MDQ6VXNlcjYyNDkyMDk3',
        login: 'kevincasey1222',
        name: 'Kevin Casey',
        node: undefined,
        teams: 'MDQ6VGVhbTQ4NTc0OTU=',
        role: 'MAINTAINER',
      },
    ]);
    expect(data.rateLimitConsumed).toBe(1);
  });
});
