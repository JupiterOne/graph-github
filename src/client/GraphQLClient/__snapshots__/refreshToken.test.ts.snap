// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pull request resources pullRequest pagination only 1`] = `
Array [
  Object {
    "additions": 2,
    "author": Object {
      "login": "kevincasey1222",
      "name": "Kevin Casey",
    },
    "authorAssociation": "MEMBER",
    "baseRefName": "main",
    "baseRefOid": "c0cdb61ee135a15d7394d2acea22b50686521f62",
    "baseRepository": Object {
      "name": "Test-repo",
      "owner": Object {
        "id": "MDEyOk9yZ2FuaXphdGlvbjg0OTIzNTAz",
        "login": "Kei-Institute",
        "url": "https://github.com/Kei-Institute",
      },
      "url": "https://github.com/Kei-Institute/Test-repo",
    },
    "body": "Here are some things which are less important, but still things, nonetheless.",
    "changedFiles": 2,
    "checksUrl": "https://github.com/Kei-Institute/Test-repo/pull/2/checks",
    "closed": false,
    "closedAt": null,
    "commits": Array [
      Object {
        "author": Object {
          "date": "2021-06-08T12:09:44-06:00",
          "user": Object {
            "login": "kevincasey1222",
          },
        },
        "authoredDate": "2021-06-08T18:09:44Z",
        "changedFiles": 1,
        "commitUrl": "https://github.com/Kei-Institute/Test-repo/commit/fccd3f6493c1117f884d4915e285a2d6263fbe7c",
        "id": "MDY6Q29tbWl0MzcxNDE5NTk4OmZjY2QzZjY0OTNjMTExN2Y4ODRkNDkxNWUyODVhMmQ2MjYzZmJlN2M=",
        "message": "Some things to seriously consider",
        "oid": "fccd3f6493c1117f884d4915e285a2d6263fbe7c",
      },
      Object {
        "author": Object {
          "date": "2021-06-08T12:11:53-06:00",
          "user": Object {
            "login": "kevincasey1222",
          },
        },
        "authoredDate": "2021-06-08T18:11:53Z",
        "changedFiles": 1,
        "commitUrl": "https://github.com/Kei-Institute/Test-repo/commit/49e7dc63370c0573e0415c1a75d5208a7a3cbeb7",
        "id": "MDY6Q29tbWl0MzcxNDE5NTk4OjQ5ZTdkYzYzMzcwYzA1NzNlMDQxNWMxYTc1ZDUyMDhhN2EzY2JlYjc=",
        "message": "Things which can be considered less seriously",
        "oid": "49e7dc63370c0573e0415c1a75d5208a7a3cbeb7",
      },
    ],
    "createdAt": "2021-06-08T18:12:39Z",
    "databaseId": 665153170,
    "deletions": 0,
    "editor": null,
    "headRefName": "anotherBranch",
    "headRefOid": "49e7dc63370c0573e0415c1a75d5208a7a3cbeb7",
    "headRepository": Object {
      "name": "Test-repo",
      "owner": Object {
        "id": "MDEyOk9yZ2FuaXphdGlvbjg0OTIzNTAz",
        "login": "Kei-Institute",
        "url": "https://github.com/Kei-Institute",
      },
    },
    "id": "MDExOlB1bGxSZXF1ZXN0NjY1MTUzMTcw",
    "isDraft": false,
    "labels": Array [
      Object {
        "id": "MDU6TGFiZWwzMDM2NTU5NTY3",
        "name": "good first issue",
        "node": undefined,
        "pullRequests": "MDExOlB1bGxSZXF1ZXN0NjY1MTUzMTcw",
      },
    ],
    "lastEditedAt": null,
    "locked": false,
    "mergeCommit": null,
    "mergeable": "MERGEABLE",
    "merged": false,
    "mergedAt": null,
    "mergedBy": null,
    "number": 2,
    "permalink": "https://github.com/Kei-Institute/Test-repo/pull/2",
    "publishedAt": "2021-06-08T18:12:39Z",
    "reviewDecision": "APPROVED",
    "reviews": Array [
      Object {
        "author": Object {
          "login": "mknoedel",
          "name": "Michael Knoedel",
        },
        "commit": Object {
          "oid": "49e7dc63370c0573e0415c1a75d5208a7a3cbeb7",
        },
        "id": "MDE3OlB1bGxSZXF1ZXN0UmV2aWV3NzQ1NjEyNTAw",
        "node": undefined,
        "pullRequests": "MDExOlB1bGxSZXF1ZXN0NjY1MTUzMTcw",
        "state": "APPROVED",
        "submittedAt": "2021-09-03T00:10:55Z",
        "updatedAt": "2021-09-03T00:10:55Z",
        "url": "https://github.com/Kei-Institute/Test-repo/pull/2#pullrequestreview-745612500",
      },
    ],
    "state": "OPEN",
    "title": "Another branch",
    "updatedAt": "2021-10-02T18:26:38Z",
    "url": "https://github.com/Kei-Institute/Test-repo/pull/2",
  },
  Object {
    "additions": 1,
    "author": Object {
      "login": "kevincasey1222",
      "name": "Kevin Casey",
    },
    "authorAssociation": "MEMBER",
    "baseRefName": "main",
    "baseRefOid": "c0cdb61ee135a15d7394d2acea22b50686521f62",
    "baseRepository": Object {
      "name": "Test-repo",
      "owner": Object {
        "id": "MDEyOk9yZ2FuaXphdGlvbjg0OTIzNTAz",
        "login": "Kei-Institute",
        "url": "https://github.com/Kei-Institute",
      },
      "url": "https://github.com/Kei-Institute/Test-repo",
    },
    "body": "This is a test pull request, with things to seriously consider.",
    "changedFiles": 1,
    "checksUrl": "https://github.com/Kei-Institute/Test-repo/pull/1/checks",
    "closed": false,
    "closedAt": null,
    "commits": Array [
      Object {
        "author": Object {
          "date": "2021-06-08T12:09:44-06:00",
          "user": Object {
            "login": "kevincasey1222",
          },
        },
        "authoredDate": "2021-06-08T18:09:44Z",
        "changedFiles": 1,
        "commitUrl": "https://github.com/Kei-Institute/Test-repo/commit/fccd3f6493c1117f884d4915e285a2d6263fbe7c",
        "id": "MDY6Q29tbWl0MzcxNDE5NTk4OmZjY2QzZjY0OTNjMTExN2Y4ODRkNDkxNWUyODVhMmQ2MjYzZmJlN2M=",
        "message": "Some things to seriously consider",
        "oid": "fccd3f6493c1117f884d4915e285a2d6263fbe7c",
      },
    ],
    "createdAt": "2021-06-08T18:10:20Z",
    "databaseId": 665150699,
    "deletions": 0,
    "editor": null,
    "headRefName": "testBranch",
    "headRefOid": "fccd3f6493c1117f884d4915e285a2d6263fbe7c",
    "headRepository": Object {
      "name": "Test-repo",
      "owner": Object {
        "id": "MDEyOk9yZ2FuaXphdGlvbjg0OTIzNTAz",
        "login": "Kei-Institute",
        "url": "https://github.com/Kei-Institute",
      },
    },
    "id": "MDExOlB1bGxSZXF1ZXN0NjY1MTUwNjk5",
    "isDraft": false,
    "labels": Array [
      Object {
        "id": "MDU6TGFiZWwzMDM2NTU5NTYw",
        "name": "documentation",
        "node": undefined,
        "pullRequests": "MDExOlB1bGxSZXF1ZXN0NjY1MTUwNjk5",
      },
      Object {
        "id": "MDU6TGFiZWwzMDM2NTU5NTY0",
        "name": "enhancement",
        "node": undefined,
        "pullRequests": "MDExOlB1bGxSZXF1ZXN0NjY1MTUwNjk5",
      },
    ],
    "lastEditedAt": null,
    "locked": false,
    "mergeCommit": null,
    "mergeable": "MERGEABLE",
    "merged": false,
    "mergedAt": null,
    "mergedBy": null,
    "number": 1,
    "permalink": "https://github.com/Kei-Institute/Test-repo/pull/1",
    "publishedAt": "2021-06-08T18:10:20Z",
    "reviewDecision": null,
    "reviews": Array [],
    "state": "OPEN",
    "title": "Some things to seriously consider",
    "updatedAt": "2021-10-02T18:26:13Z",
    "url": "https://github.com/Kei-Institute/Test-repo/pull/1",
  },
]
`;
